src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -255,6 +256,62 @@ public Icon getSystemIcon(File f) {
         }
     }
 
+   /**
+    * Returns an icon for a file, directory, or folder as it would be displayed
+    * in a system file browser for the requested size.
+    * <p>
+    * Example: <pre>
+    *     FileSystemView fsv = FileSystemView.getFileSystemView();
+    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64, 64);
+    *     JLabel label = new JLabel(icon);
+    * </pre>
+    *
+    * @implSpec The available icons may be platform specific and so the
+    * available sizes determined by the platform. Therefore an exact match
+    * for the requested size may not be possible.
+    *
+    * The icon returned may be a multi-resolution icon image,
+    * which allows better support for High DPI environments
+    * with different scaling factors.
+    *
+    * @param f a {@code File} object for which the icon will be retrieved
+    * @param width width of the icon in user coordinate system.
+    * @param height height of the icon in user coordinate system.
+    * @return an icon as it would be displayed by a native file chooser
+    * or null if invalid parameters are passed such as reference to a
+    * non-existing file.
+    * @see JFileChooser#getIcon
+    * @see AbstractMultiResolutionImage
+    * @see FileSystemView#getSystemIcon(File)
+    * @since 17
+    */
+    public Icon getSystemIcon(File f, int width, int height) {
+        if (height <1 || width < 1) {
+            throw new IllegalArgumentException("Icon size can not be below 1");
+        }
+
+        if (f == null || !f.exists()) {
+            return null;
```

Comment:
I suggested the a null File ought to be IAE too - this is showing up in the conversation thread but not here.